cmd/infra/aws/util/util.go

```diff
@@ -31,6 +32,53 @@ func NewSession(agent string, credentialsFile string, credKey string, credSecret
 	return awsSession
 }
 
+func NewStsSession(agent, credentialsFile, credKey, credSecretKey, sessionToken, roleArn, region string) (*session.Session, error) {
+	stsSessionOpts := session.Options{}
+
+	if credKey != "" && credSecretKey != "" && sessionToken != "" {
+		stsSessionOpts.Config.Credentials = credentials.NewStaticCredentials(credKey, credSecretKey, sessionToken)
+	}
+
+	if credentialsFile != "" {
+		stsSessionOpts.SharedConfigFiles = append(stsSessionOpts.SharedConfigFiles, credentialsFile)
```

Comment:
this is expecting an aws credentials file, I think it should expect the output of `aws sts get-session-token --output json` instead and then parse it and extract necessary fields.
Otherwise, the use will have to do 1 extra step to format the sts-creds file before passing it to the cli. If we do this, we can validate that the session_token exist and throw an error/warning that the user is not using temporary credentials.